### ISSUE TITLE
Disallow @Regex qualifier on non-String types

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#122]
```
The Regex checker allows qualifiers on any Java types, not just Strings.  The 
attached program shows how one can qualify basic types like boolean and object 
types like Boolean, which do not make sense as regular expressions.

I'll assign this to Eric, since he has the most recent experience with the 
Regex checker.

What steps will reproduce the problem?
1. Compile attached program using checkers.regex.RegexChecker
2. Observe that no errors are reported.

What is the expected output? What do you see instead?

It would be better to issue an error for any annotation on a non-String type.


```

Original issue reported on code.google.com by `charlie....@gmail.com` on 13 Jan 2012 at 10:55

Attachments:
* [NonStringRegex.java](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-122/comment-0/NonStringRegex.java)
